sw/device/silicon_creator/lib/drivers/spi_device.h

```diff
@@ -10,6 +10,8 @@
 
 #include "sw/device/silicon_creator/lib/error.h"
 
+#include "spi_device_regs.h"
```

Comment:
We usually try to avoid exposing the device register header file at the driver interface.  We add a bit of duplicate code by setting the enum here with hardcoded values, and then we make sure that the value is consistent with the register definition by using a `static_assert` in the `.c` file.

Reply:
Ah, I was unsure of the convention here, thanks @moidx! @nmosier see here for an example: https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/lib/dif/dif_alert_handler.c;drc=cdd49323dff06023ead342c9e71f1c75a0be72e9;l=14